### PR TITLE
Add `httpx` support

### PR DESCRIPTION
**Pull request by vtbassmatt**
Fixes https://github.com/daggaz/json-stream/issues/27

### CODE REVIEW

**Comment by vtbassmatt** [comment]
I tried to make each commit reviewable on its own, though the total PR is relatively readable, too.

The duplication between the `httpx` and `requests` modules (and their tests) feels unfortunate. If you're open to renaming the `requests` module something more generic, then we could do something like `getattr(response, 'iter_content')` vs `getattr(response, 'iter_bytes')` to see which kind of `Response` we're dealing with. But I don't feel at all strongly about the topic.

File: README.md [hunk 1]
```diff
@@ -220,6 +221,12 @@ z at path ('xxxx', 3)
 
 ### Stream a URL
 
+`json_stream` knows how to stream directly from a URL using a variety of packages.
+Supported packages include:
+- Python's batteries-included `urllib` package
```
Comment (daggaz) [inline README.md:226]: link to docs?

File: README.md [hunk 2]
```diff
@@ -220,6 +221,12 @@ z at path ('xxxx', 3)
...
+- The popular [`requests`](https://pypi.org/project/httpx/) library
```
Comment (daggaz) [inline README.md:227]: wrong URL

File: pyproject.toml [hunk 3]
```diff
@@ -35,7 +35,10 @@ tracker = "https://github.com/daggaz/json-stream/issues"
 funding = "https://www.buymeacoffee.com/daggaz"
 
 [project.optional-dependencies]
-requests = ["requests"]
+requests = [
```
Comment (daggaz) [inline pyproject.toml:38]: shouldn't this be a second optional called `httpx`?

**Review by daggaz — CHANGES_REQUESTED** [review]

**Comment by daggaz** [comment]
I agree that the duplication between the modules is unfortunate.

In theory you could factor out the "thing that gives you an interable" (specific to requests/httpx) from the "thing that processes an iterable" (maybe change the interface to `json_stream.load()` to accept a file _or_ an iterable and wrap appropriately. 

But I'm happy to accept this PR as is (pending the resolution of my comments) and I can roll up the refactoring work into some bigger plans I have (related to async support).

File: pyproject.toml (hunk 3 above)
Comment (vtbassmatt) [reply pyproject.toml:38]: Good catch. I don't know much about pyproject.toml (except what `poetry` automatically writes) but this is clearly boneheaded.

File: README.md (hunk 2 above)
Comment (vtbassmatt) [reply README.md:227]: 🤦 thank you

**Review by daggaz — APPROVED** [review]

**Pull request merged**